`make test` is failing

```
dep package is available
gometalinter package is available

Running test ./test/001-testtest.sh
Running test ./test/020-gofmt.sh
./gluster-exporter/main.go does not conform to gofmt rules. Run: gofmt -s -w ./gluster-exporter/main.go
./gluster-exporter/conf/conf.go does not conform to gofmt rules. Run: gofmt -s -w ./gluster-exporter/conf/conf.go
Running test ./test/020-golint.sh
./gluster-exporter/conf/conf.go:10:5: exported var Collectors should have comment or be unexported
./gluster-exporter/conf/conf.go:11:5: exported var SystemConfig should have comment or be unexported
./gluster-exporter/conf/conf.go:13:6: exported type CollectorConfig should have comment or be unexported
./gluster-exporter/conf/conf.go:19:1: exported function LoadCollectorsConfig should have comment or be unexported
./gluster-exporter/conf/conf.go:38:1: exported function LoadConfig should have comment or be unexported
Found 5 lint suggestions; failing.
Running test ./test/020-gometalinter.sh
Running test ./test/020-test-makefiles.sh
Running test ./test/020-test-shellscripts.sh
warning: could not find shellcheck ... will skip checks
Running test ./test/030-govet.sh
# github.com/gluster/gluster-prometheus/gluster-exporter/conf
../go/src/github.com/gluster/gluster-prometheus/gluster-exporter/conf/conf.go:39:26: undefined: confFilePath
../go/src/github.com/gluster/gluster-prometheus/gluster-exporter/conf/conf.go:41:56: undefined: confFilePath
../go/src/github.com/gluster/gluster-prometheus/gluster-exporter/conf/conf.go:47:70: undefined: confFilePath
Running test ./test/100-gotest.sh
# github.com/gluster/gluster-prometheus/gluster-exporter/conf
../go/src/github.com/gluster/gluster-prometheus/gluster-exporter/conf/conf.go:39:26: undefined: confFilePath
../go/src/github.com/gluster/gluster-prometheus/gluster-exporter/conf/conf.go:41:56: undefined: confFilePath
../go/src/github.com/gluster/gluster-prometheus/gluster-exporter/conf/conf.go:47:70: undefined: confFilePath
ERROR: failing tests:
  ./test/020-gofmt.sh
  ./test/020-golint.sh
  ./test/030-govet.sh
  ./test/100-gotest.sh
make: *** [Makefile:85: test] Error 1
```

--- a/gluster-exporter/conf/collectors.conf.sample
+++ b/gluster-exporter/conf/collectors.conf.sample
@@ -1,0 +1,12 @@
+[
+	{
+		"name": "gluster_ps",
+		"sync_interval": 5,
+		"disabled": false
+	},
+	{
+		"name": "gluster_brick",
+		"sync_interval": 5,
+		"disabled": false
+	}
+]

--- a/gluster-exporter/conf/conf.go
+++ b/gluster-exporter/conf/conf.go
@@ -1,0 +1,57 @@
+package conf
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+// Collectors maintain the global collectors configurations
+var Collectors = make(map[string]CollectorConfig)
+
+// SystemConfig maintains the global system configurations
+var SystemConfig = make(map[string]string)
+
+// CollectorConfig struct defines the structure of collectors configuration
+type CollectorConfig struct {
+	Name         string `json:"name"`
+	SyncInterval uint64 `json:"sync_interval"`
+	Disabled     bool   `json:"disabled"`
+}
+
+// LoadCollectorsConfig loads the collectors configuration
+func LoadCollectorsConfig(confFilePath string) string {
+	fileObj, err := os.Open(confFilePath)
+	if err != nil {
+		msg := fmt.Sprintf("Failed loading config file: %s", confFilePath)
+		return msg
+	}
+	defer fileObj.Close()
+	data, _ := ioutil.ReadAll(fileObj)
+	var confs []CollectorConfig
+	if err := json.Unmarshal(data, &confs); err != nil {
+		msg := fmt.Sprintf("Error parsing the content of config file: %s", confFilePath)
+		return msg
+	}
+	for _, conf := range confs {
+		Collectors[conf.Name] = conf
+	}
+	return ""
+}
+
+// LoadConfig loads the global system configuration
+func LoadConfig(confFilePath string) string {
+	fileObj, err := os.Open(confFilePath)
+	if err != nil {
+		msg := fmt.Sprintf("Failed loading config file: %s", confFilePath)
+		return msg
+	}
+	defer fileObj.Close()
+	data, _ := ioutil.ReadAll(fileObj)
+	if err := json.Unmarshal(data, &SystemConfig); err != nil {
+		msg := fmt.Sprintf("Error parsing the content of config file: %s", confFilePath)
+		return msg
+	}
+	return ""
+}

--- a/gluster-exporter/conf/global.conf.sample
+++ b/gluster-exporter/conf/global.conf.sample
@@ -1,0 +1,8 @@
+{
+	"gluster-mgmt": "glusterd1",
+	"glusterd-dir": "/var/lib/glusterd/",
+	"gluster-binary-path": "gluster",
+	"gd2-rest-endpoint": "http://192.168.10.11:24007",
+	"port": "8080",
+	"metrics-path": "/metrics"
+}

--- a/gluster-exporter/main.go
+++ b/gluster-exporter/main.go
@@ -9,9 +9,11 @@ import (
 	"os"
 	"regexp"
 	"runtime"
+	"strconv"
 	"strings"
 	"time"
 
+	"github.com/gluster/gluster-prometheus/gluster-exporter/conf"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
@@ -24,12 +26,10 @@ var (
 	defaultGlusterd2Workdir = ""
 )
 var (
-	glusterMgmt     = flag.String("gluster-mgmt", "glusterd1", "Choice of GlusterD version i.e glusterd1 or glusterd2, Default is glusterd1")
-	glusterdWorkdir = flag.String("glusterd-dir", "", "Directory where the local peer info file is stored, Default for glusterd1 is /var/lib/glusterd/ and for glusterd2 is /var/lib/glusterd2/")
-	port            = flag.Int("port", 8080, "Exporter Port")
-	metricsPath     = flag.String("metrics-path", "/metrics", "Metrics API Path")
-	volinfo         = flag.String("volinfo", "", "Volume info json file")
-	showVersion     = flag.Bool("version", false, "Show the version information")
+	volinfo       = flag.String("volinfo", "", "Volume info json file")
+	showVersion   = flag.Bool("version", false, "Show the version information")
+	config        = flag.String("config", "", "Global config file path")
+	collectorsCfg = flag.String("collectors-config", "", "Collectors config file path")
 
 	peerID          string
 	defaultInterval time.Duration = 5
@@ -37,33 +37,23 @@ var (
 )
 
 type glusterMetric struct {
-	name            string
-	fn              func()
-	intervalSeconds time.Duration
+	name string
+	fn   func()
 }
 
 var glusterMetrics []glusterMetric
 
-func registerMetric(name string, fn func(), intervalSeconds int64) {
-	glusterMetrics = append(glusterMetrics, glusterMetric{name: name, fn: fn, intervalSeconds: time.Duration(intervalSeconds)})
-}
-
-func getGlusterdWorkdir() string {
-	if *glusterdWorkdir != "" {
-		return *glusterdWorkdir
-	}
-	if *glusterMgmt == "glusterd2" {
-		return defaultGlusterd2Workdir
-	}
-	return defaultGlusterd1Workdir
+func registerMetric(name string, fn func()) {
+	glusterMetrics = append(glusterMetrics, glusterMetric{name: name, fn: fn})
 }
 
 func getPeerID() (string, error) {
 	if peerID == "" {
-		workdir := getGlusterdWorkdir()
+		glusterMgmt, _ := conf.SystemConfig["gluster-mgmt"]
+		workdir, _ := conf.SystemConfig["glusterd-dir"]
 		keywordID := "UUID"
 		filepath := workdir + "/glusterd.info"
-		if *glusterMgmt == "glusterd2" {
+		if glusterMgmt == "glusterd2" {
 			keywordID = "peer-id"
 			filepath = workdir + "/uuid.toml"
 		}
@@ -107,6 +97,15 @@ func dumpVersionInfo() {
 func main() {
 	flag.Parse()
 
+	if err := conf.LoadCollectorsConfig(*collectorsCfg); err != "" {
+		fmt.Fprintf(os.Stderr, "Loading collectors config failed: %s", err)
+		os.Exit(1)
+	}
+	if err := conf.LoadConfig(*config); err != "" {
+		fmt.Fprintf(os.Stderr, "Loading global config failed: %s", err)
+		os.Exit(1)
+	}
+
 	if *showVersion {
 		dumpVersionInfo()
 		return
@@ -115,16 +114,20 @@ func main() {
 	// start := time.Now()
 
 	for _, m := range glusterMetrics {
-		go func(m glusterMetric) {
-			for {
-				m.fn()
-				interval := defaultInterval
-				if m.intervalSeconds > 0 {
-					interval = m.intervalSeconds
-				}
-				time.Sleep(time.Duration(time.Second * time.Duration(interval)))
+		if collectorConf, ok := conf.Collectors[m.name]; ok {
+			if collectorConf.Disabled == false {
+				go func(m glusterMetric) {
+					for {
+						m.fn()
+						interval := defaultInterval
+						if collectorConf.SyncInterval > 0 {
+							interval = time.Duration(collectorConf.SyncInterval)
+						}
+						time.Sleep(time.Duration(time.Second * time.Duration(interval)))
+					}
+				}(m)
 			}
-		}(m)
+		}
 	}
 
 	if len(glusterMetrics) == 0 {
@@ -132,8 +135,11 @@ func main() {
 		os.Exit(1)
 	}
 
-	http.Handle(*metricsPath, promhttp.Handler())
-	err := http.ListenAndServe(fmt.Sprintf(":%d", *port), nil)
+	metricsPath, _ := conf.SystemConfig["metrics-path"]
+	port, _ := conf.SystemConfig["port"]
+	nport, _ := strconv.Atoi(port)
+	http.Handle(metricsPath, promhttp.Handler())
+	err := http.ListenAndServe(fmt.Sprintf(":%d", nport), nil)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Failed to run exporter\nError: %s", err)
 		os.Exit(1)

--- a/gluster-exporter/metric_brick.go
+++ b/gluster-exporter/metric_brick.go
@@ -235,5 +235,5 @@ func init() {
 
 	// Register to update this every 2 seconds
 	// Name, Callback Func, Interval Seconds
-	registerMetric("gluster_brick", brickUtilization, 5)
+	registerMetric("gluster_brick", BrickUtilization)
 }

--- a/gluster-exporter/metric_ps.go
+++ b/gluster-exporter/metric_ps.go
@@ -232,5 +232,5 @@ func init() {
 
 	// Register to update this every 2 seconds
 	// Name, Callback Func, Interval Seconds
-	registerMetric("gluster_ps", ps, 2)
+	registerMetric("gluster_ps", ps)
 }

